daemon/api_aspects_test.go

```diff
@@ -0,0 +1,320 @@
+	} {
+		restore := daemon.MockAspectstateGet(func(_ *state.State, acc, bundleName, aspect, field string, value interface{}) error {
+			return t.err
+		})
+
+		buf := bytes.NewBufferString(`{"account": "system", "bundle": "network", "aspect": "wifi-setup", "field": "ssid"}`)
+		req, err := http.NewRequest("GET", "/v2/aspects", buf)
+		req.Header.Set("Content-Type", "application/json")
+		c.Assert(err, IsNil, Commentf("%s test", t.name))
+
+		rspe := s.errorReq(c, req, nil)
+		c.Check(rspe.Status, Equals, t.code, Commentf("%s test", t.name))
+		restore()
+	}
+}
+
+func (s *aspectsSuite) TestGetAspectMissingField(c *C) {
+	s.daemon(c)
+
+	type test struct {
+		acc    string
+		bundle string
+		aspect string
+		field  string
+	}
+
+	for _, t := range []test{
+		{bundle: "a", aspect: "b", field: "c"},
+		{acc: "a", aspect: "b", field: "c"},
+		{acc: "a", bundle: "b", field: "c"},
+		{acc: "a", bundle: "b", aspect: "c"},
+	} {
+		buf := bytes.NewBufferString(fmt.Sprintf(`{"account": %q, "bundle": %q, "aspect": %q, "field": %q}`, t.acc, t.bundle, t.aspect, t.field))
+		req, err := http.NewRequest("GET", "/v2/aspects", buf)
+		req.Header.Set("Content-Type", "application/json")
+		c.Assert(err, IsNil)
+
+		rspe := s.errorReq(c, req, nil)
+		c.Check(rspe.Status, Equals, 400)
+	}
+}
+
+func (s *aspectsSuite) TestGetAspectMalformedRequest(c *C) {
+	s.daemon(c)
+
+	buf := bytes.NewBufferString(`{`)
+	req, err := http.NewRequest("GET", "/v2/aspects", buf)
+	req.Header.Set("Content-Type", "application/json")
+	c.Assert(err, IsNil)
+
+	rspe := s.errorReq(c, req, nil)
+	c.Check(rspe.Status, Equals, 400)
+	c.Check(rspe.Message, Equals, "cannot decode aspect request body: unexpected EOF")
+}
+
+func (s *aspectsSuite) TestSetAspect(c *C) {
+	s.daemon(c)
+
+	type test struct {
+		name  string
+		value interface{}
+	}
+
+	for _, t := range []test{
+		{name: "string", value: "foo"},
+		{name: "integer", value: float64(123)},
+		{name: "list", value: []interface{}{"foo", "bar"}},
+		{name: "map", value: map[string]interface{}{"foo": "bar"}},
+	} {
+		cmt := Commentf("%s test", t.name)
+		restore := daemon.MockAspectstateSet(func(_ *state.State, acc, bundleName, aspect, field string, value interface{}) error {
+			c.Check(acc, Equals, "system", cmt)
+			c.Check(bundleName, Equals, "network", cmt)
+			c.Check(aspect, Equals, "wifi-setup", cmt)
+			c.Check(field, Equals, "ssid", cmt)
+			c.Check(value, DeepEquals, t.value, cmt)
+			return nil
+		})
+		jsonVal, err := json.Marshal(t.value)
+		c.Check(err, IsNil, cmt)
+
+		buf := bytes.NewBufferString(fmt.Sprintf(`{"account": "system", "bundle": "network", "aspect": "wifi-setup", "field": "ssid", "value": %s}`, jsonVal))
+		req, err := http.NewRequest("POST", "/v2/aspects", buf)
+		req.Header.Set("Content-Type", "application/json")
+		c.Check(err, IsNil, cmt)
+
+		rspe := s.syncReq(c, req, nil)
+		c.Check(rspe.Status, Equals, 200, cmt)
+		restore()
+	}
+}
+
+func (s *aspectsSuite) TestUnsetAspect(c *C) {
```

Comment:
I do not think the concept of "unset" is mentioned in the specification. Consider adding it.